src/fs.c

```diff
@@ -373,8 +371,8 @@ static void luv_fs_cb(uv_fs_t* req) {
     lua_insert(L, -nargs - 1);
     nargs++;
   }
-  luv_fulfill_req(L, (luv_req_t*)req->data, nargs);
-  LUV_FS_CLEANUP_REQ
+  luv_fulfill_req(L, data, nargs);
+  luv_fs_cleanup_req(L, req, data);
```

Comment:
This is not going to work. `uv_fs_req_cleanup` has to be called before `luv_fulfill_req` to avoid the race condition.